### PR TITLE
Load generated sources from ballerina project

### PR DESCRIPTION
**Pull request by Dilhasha**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/38766

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java [hunk 1]
```diff
@@ -138,15 +138,31 @@ private Optional<Path> modulePath(ModuleId moduleId) {
         return Optional.empty();
     }
 
+    private Optional<Path> generatedModulePath(ModuleId moduleId) {
+        if (currentPackage().moduleIds().contains(moduleId)) {
+            if (currentPackage().getDefaultModule().moduleId() == moduleId) {
+                return Optional.of(sourceRoot);
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java:144]: Shouldn't this return `sourceRoot.resolve(ProjectConstants.GENERATED_MODULES_ROOT)`?

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java [hunk 2]
```diff
@@ -138,15 +138,31 @@ private Optional<Path> modulePath(ModuleId moduleId) {
...
+            Optional<Path> generatedModulePath = generatedModulePath(moduleId);
             if (module.documentIds().contains(documentId)) {
-                if (modulePath.isPresent()) {
-                    return Optional.of(modulePath.get().resolve(module.document(documentId).name()));
+                if (generatedModulePath.isPresent()) {
+                    if (Files.exists(generatedModulePath.get().resolve(module.document(documentId).name()))) {
+                        return Optional.of(generatedModulePath.get().resolve(module.document(documentId).name()));
+                    }
                 }
+                return Optional.of(modulePath.get().resolve(module.document(documentId).name()));
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java:165]: Shouldn't we keep the check at L147?

Comment (azinneera) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java:165]: Shall we keep the existing checks as it and add another else-if for generated directory after L172?

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java (hunk 1 above)
Comment (Dilhasha) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java:144]: yes. I'll correct.

**Review by azinneera — APPROVED** [review]

**Pull request merged**